VideoLocker/src/main/java/org/edx/mobile/view/CourseDiscussionPostsThreadFragment.java

```diff
@@ -171,6 +172,24 @@ public void onDestroy() {
         EventBus.getDefault().unregister(this);
     }
 
+    @Override
+    public void onRestart() {
+        if (postsSort == DiscussionPostsSort.LAST_ACTIVITY_AT && mSelectedItem != -1) {
+            // Move the last viewed thread to the top
+            discussionPostsAdapter.moveToTop(mSelectedItem);
+            mSelectedItem = -1;
+        }
+
+        nextPage = 1;
+        controller.refresh();
```

Comment:
Makes sense, I guess. We should add pull to refresh functionality at some point though, at which point my suggested change can be implemented.